lib/mongoid/association/nested/one.rb

```diff
@@ -32,6 +32,8 @@ def build(parent)
             parent.send(association.setter, Factory.build(@class_name, attributes))
           elsif delete?
             parent.send(association.setter, nil)
+          else
+            check_for_id_violation!
```

Comment:
Maybe rename to `check_for_id_mutation!` or `id_change!`? "Violation" in this context sounds like a malformed (i.e. non-BSON) ID. Similar with `acceptable_id` ID below.